### PR TITLE
Fix j9 crash by avoiding instantiating WrappedString

### LINKED ISSUE #4985: [1.3.0-RC4] Error when launching sbt on AdoptOpenJDK OpenJ9
## steps
sbt version: *1.3.0-RC4*

Start sbt using AdoptOpenJDK J9 on Linux

## problem

When starting sbt 1.3.0-RC4 on Ubuntu or WSL (Debian) with AdoptOpenJDK J9, a stacktrace is printed almost right away: https://gist.github.com/hugo-vrijswijk/c49fbbdc613a088efb7819a02957c36f

3 files are also written to disk:

https://gist.github.com/hugo-vrijswijk/e284188579cf729ce80389792f79e4d9

And a binary file with the name `Snap.20190824.112919.24863.0003.trc` I could upload somewhere else if needed.

## expectation

Sbt starts normally.

## notes
WSL: 
- Java versions: AdoptOpenJDK J9 8.0.222 or 11.0.4
- Used latest `sbt-extras` script
- Distro: Debian 10
- Windows version 1903

Ubuntu:
- Java version: AdoptOpenJDK J9 11.0.4
- Used latest `sbt-extras` script
- Distro: Ubuntu 18.0.4

Works fine when in the following cases:
- OpenJDK 11 WSL
- AdoptOpenJDK 11 HotSpot on WSL
- AdoptOpenJDK OpenJ9 11 on Windows.

### PR DESCRIPTION
**Pull request by eatkins**
https://github.com/sbt/sbt/issues/4985 reported that sbt would
immediately crash when using the latest launcher (1.1.2). The crashes
indicated that the problems arose when jit-ing
scala/collection/immutable/WrappedString.apply. We can avoid making a
WrappedString by using `+:` instead of `++`.

Fixes https://github.com/sbt/sbt/issues/4985.

### CODE REVIEW

File: launcher-implementation/src/main/scala/xsbt/boot/Launch.scala [hunk 1]
```diff
@@ -228,11 +228,11 @@ class Launch private[xsbt] (val bootDirectory: File, val lockBoot: Boolean, val
       // When sbt core artifacts are not properly downloaded, this the point that fails with "No Scala version specified or detected"
       val scalaVersion = getOrError(
         strictOr(explicitScalaVersion, retrievedApp.detectedScalaVersion),
-        s"""sbt/sbt#4955!
+        (s"""sbt/sbt#4955!
            |sbt launcher is unable to detect the Scala version for ${id.groupID}:${id.name};
            |this likely indicates an incomplete artifact resolution and/or corrupt boot cache (~/.sbt/boot/).
            |the following is the full classpath that was retrieved for ${id.groupID}:${id.name}:
-           |""".stripMargin ++ retrievedApp.fullClasspath.toList.map(x => " * " + x.getAbsolutePath).mkString(System.lineSeparator)
+           |""".stripMargin +: retrievedApp.fullClasspath.toList.map(x => " * " + x.getAbsolutePath)).mkString(System.lineSeparator)
```
Comment (eed3si9n) [inline launcher-implementation/src/main/scala/xsbt/boot/Launch.scala]: This looks like I originally typo'ed or got confused by the code behind it, but shouldn't it be just straight-up `+` to concatenate two strings?

**Review by eed3si9n — CHANGES_REQUESTED** [review]

**Comment by eatkins** [comment]
Oh, yeah, that’s true `+` vs `++` would probably fix the issue. While I’m at it, do we really want the sbt issue number in the message?

**Comment by eed3si9n** [comment]
> While I’m at it, do we really want the sbt issue number in the message?

Yes, I think this is a good idea, if I say so myself. The number is unique, and if people run into this issue, (which should not normally happen unless your cache is messed up) they can comment on it. I wish I could put GitHub issue reference to all error messages.

**Comment by eatkins** [comment]
I'm not as enthusiastic about having issues in the error messages, but I don't mind.

```
> console
[info] Starting scala interpreter...
[info]
Welcome to Scala version 2.10.7 (OpenJDK 64-Bit Server VM, Java 1.7.0_222).
Type in expressions to have them evaluated.
Type :help for more information.

scala> "foo" + "bar"
res0: String = foobar

scala> "foo" ++ "bar"
res1: String = foobar
```

I'm not a huge fan of WrappedString being in the default scope. It leads to a lot of confusing behavior like:
```
scala> Seq("foo", "bar").flatMap(_ + "baz")
res2: Seq[Char] = List(f, o, o, b, a, z, b, a, r, b, a, z)
```
(Obviously that's a scala thing not an sbt thing).

File: launcher-implementation/src/main/scala/xsbt/boot/Launch.scala (hunk 1 above)
Comment (viktorklang) [reply launcher-implementation/src/main/scala/xsbt/boot/Launch.scala]: why not `.iterator.map … mkString`?

Comment (eed3si9n) [reply launcher-implementation/src/main/scala/xsbt/boot/Launch.scala]: I guess because I prefer concrete datatype until it matters.

**Pull request merged**